### DISCUSSION TITLE
My fish shell always goto / everytime luanch , how to fix it ?

### DISCUSSION THREAD

**Discussion by q2333gh** [fish-shell/fish-shell#10311] [Q&A] [answered]
```bash
btwl@btwl-virtual-machine ~/.c/fish> echo $SHELL
/usr/bin/fish
btwl@btwl-virtual-machine ~/.c/fish> pwd
/home/btwl/.config/fish
btwl@btwl-virtual-machine ~/.c/fish> fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
btwl@btwl-virtual-machine /> pwd
/     
```
why my fish will behave like this goto root  when i run fish ? which confuse me . i want it behave like default.

**Comment by q2333gh** [comment]
I tried this function in config.fish . but not work.
```
if status is-login
    cd $PWD
    # cd ~
end
```

**Comment by krobelus** [comment]
if it works in `HOME=$(mktemp -d ) fish` then it's a problem in your config.
Run `fish_trace=1 fish -i -c exit`  and look for ` cd ` to see where it's run.

**Reply by q2333gh** [reply]
As  i test some cmd in different user. i found that only the  user `btwl` account will cause this redirect problem
```bash
Type help for instructions on how to use fish
btwl@btwl-virtual-machine /> cd /home/btwl/code/ic/tax_lint
btwl@btwl-virtual-machine ~/c/i/tax_lint (master)> sudo fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
root@btwl-virtual-machine /h/b/c/i/tax_lint (master)# fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
root@btwl-virtual-machine /h/b/c/i/tax_lint (master)# su btwl
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
btwl@btwl-virtual-machine /> 

```

Maybe i can try to clear all personal config of fish of user btwl ?

**Comment by matoruru** [comment, marked as answer]
@q2333gh 

Don't you have any special config in those files?

- `~/.config/fish/config.fish`
- `~/.config/fish/conf.d/*`

Only that user `btwl` has something special in there.

**Reply by q2333gh** [reply]
After completely comment all content in  `~/.config/fish/config.fish`  it finally solved! Now i can find out which detailed config cause this problem hah. 
Thanks @matoruru !

**Reply by q2333gh** [reply]
It turns out very subtle problem here !!! 😆 

lets say if my  config.fish got this syntax error 
```
#  disable vpn 
function dvpn
    export https_proxy= && export https_proxy=
end
/    # and i have the stupid / symbol right here .   It cause  redirect bug ! 
```

**Reply by matoruru** [reply]
Just 1 letter caused this issue! Good catch😀
